File: tessoku/A07 - Event Attendance/A07.ts
const mainA07 = (input: string): void => {
    const lines: string[] = input.trim().split('\n');
    const D: number = parseInt(lines[0]);
    const N: number = parseInt(lines[1]);
    
    const imos: number[] = new Array(D + 1).fill(0);
    
    for (let i = 0; i < N; i++) {
        const [L, R]: number[] = lines[i + 2].split(' ').map(Number);
        imos[L - 1]++;
        imos[R]--;
    }
    
    for (let day = 0; day < D; day++) {
        if (day > 0) {
            imos[day] += imos[day - 1];
        }
        console.log(imos[day]);
    }
}

mainA07(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: tessoku/A07 - Event Attendance/A07.cs
using System;
using System.Linq;

class Program
{
   static void Main()
   {
       int D = int.Parse(Console.ReadLine());
       int N = int.Parse(Console.ReadLine());

       int[] imos = new int[D + 1];

       for (int i = 0; i < N; i++)
       {
           int[] input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
           int L = input[0];
           int R = input[1];

           imos[L - 1]++;
           imos[R]--;
       }

       for (int day = 0; day < D; day++)
       {
           if (day > 0)
           {
               imos[day] += imos[day - 1];
           }
           Console.WriteLine(imos[day]);
       }
   }
}